andela/Ah-backend-aquaman
# #164069218 send verification email after successful signup

### What does this PR do?

- add EmailVerifyAPIView
- implement token decoding
- add URL for EmailVerifyAPIView
- write tests for the EmailVerifyAPIView

### Description of Task to be completed?
- Users should receive an email containing an activation link when they successfully signup.
- Users' accounts should be activated when they click on the activation link they received in their email

#### How should this be manually tested?
```
- clone the repo
- cd Ah-backend-aquaman
- make a virtual environment and activate it
- install packages with pip install -r requirements.txt
- run python manage.py runserver
- create an account using a rest client by hitting POST /api/users/
- Try signing in by hitting POST /api/users/login/
- After it fails, go to your mailbox and click on the link to verify
- Then Try signing in again
```


### Any background context you want to provide?
With email verification, we prevent malicious bots from creating random accounts and creating DOS attacks on our server.

### What are the relevant pivotal tracker stories?
[#164069218](https://www.pivotaltracker.com/story/show/164069218)
